src/cmake/modules/FindLibheif.cmake

```diff
@@ -30,7 +30,12 @@ find_library (LIBHEIF_LIBRARY heif
               DOC "The directory where libheif libraries reside")
 
 if (LIBHEIF_INCLUDE_DIR)
-    file(STRINGS "${LIBHEIF_INCLUDE_DIR}/libheif/heif_version.h" TMP REGEX "^#define LIBHEIF_VERSION[ \t].*$")
+    execute_process(
+        COMMAND ${CMAKE_CXX_COMPILER} -E -dD -P "${LIBHEIF_INCLUDE_DIR}/libheif/heif_version.h"
```

Comment:
Does this do the right thing with all compilers? MSVS?

Reply:
No, if MSVS, sould ``set(PREPROCESS_COMMAND "${CMAKE_CXX_COMPILER} /E ${LIBHEIF_INCLUDE_DIR}/libheif/heif_version.h")``

Maybe I should change the code to something like this?

```bash
    if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
        # msvc
        set(PREPROCESS_COMMAND "${CMAKE_CXX_COMPILER} /E ${LIBHEIF_INCLUDE_DIR}/libheif/heif_version.h")
    else ()
        # icc, gcc or clang
        set(PREPROCESS_COMMAND "${CMAKE_CXX_COMPILER} -E -dD -P ${LIBHEIF_INCLUDE_DIR}/libheif/heif_version.h")
    endif ()
```